dataland-backend/src/main/kotlin/org/dataland/datalandbackend/service/DataManager.kt

```diff
@@ -0,0 +1,156 @@
+package org.dataland.datalandbackend.service
+
+import org.dataland.datalandbackend.edcClient.api.DefaultApi
+import org.dataland.datalandbackend.interfaces.DataManagerInterface
+import org.dataland.datalandbackend.model.CompanyMetaInformation
+import org.dataland.datalandbackend.model.DataManagerInputToGetData
+import org.dataland.datalandbackend.model.DataMetaInformation
+import org.dataland.datalandbackend.model.StorableDataSet
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * Implementation of a data manager for Dataland including meta data storages
+ */
+@Component("DefaultManager")
+class DataManager(
+    @Autowired var edcClient: DefaultApi
+) : DataManagerInterface {
+    var dataMetaData = mutableMapOf<String, DataMetaInformation>()
+    var companyData = mutableMapOf<String, CompanyMetaInformation>()
+    private var companyCounter = 0
+
+    /*
+    ________________________________
+    Helper-methods:
+    ________________________________
+     */
+
+    private fun verifyCompanyIdExists(companyId: String) {
+        if (!companyData.containsKey(companyId)) {
+            throw IllegalArgumentException("Dataland does not know the company ID $companyId.")
+        }
+    }
+
+    private fun verifyDataIdIsExists(dataId: String) {
+        if (!dataMetaData.containsKey(dataId)) {
+            throw IllegalArgumentException("Dataland does not know the data ID: $dataId.")
+        }
+    }
+
+    private fun verifyDataTypeExists(dataType: String) {
+        val matchesForDataType = dataMetaData.filter { it.value.dataType.equals(dataType) }
+        if (matchesForDataType.isEmpty()) {
+            throw IllegalArgumentException("Dataland does not know the data type: $dataType")
+        }
+    }
+
+    /*
+    ________________________________
+    Methods to route data inserts and queries to the data store and save meta data in the Dataland-Meta-Data-Storage:
+    ________________________________
+     */
+
+    override fun addDataSet(storableDataSet: StorableDataSet): String {
+        verifyCompanyIdExists(storableDataSet.companyId)
+
+        val dataId = edcClient.insertData(storableDataSet.data)
+
+        dataMetaData[dataId] =
```

Comment:
Wir vertrauen drauf, dass EuroDaT keine dataIds doppelt zurückgeben wird. Sollten wir das aber nicht trotzdem überprüfen? 
Sprich: If `dataMetaData.containsKey(dataId) { throw SomeException }`

Reply:
Habe ich so implementiert:

![image](https://user-images.githubusercontent.com/99255296/161248953-afa6e4b0-45a4-4061-b923-e5ad6128da01.png)

Aber sollten wir diese Exception nicht dann auch catchen, um zu einem sauberen Stand zurückzukehren? Möglichkeiten:
- Catch => den Datensatz in EuroDat wieder löschen (quasi Rollback)
- Catch => die DataId bei uns in Dataland abweichend definieren. z.B. wenn 500 bereits in Dataland existiert, EuroDAt aber 500 zurpckgibt, dann in Dataland die ID unter "EuroDaT-dataId-500" speicher o.Ä. ?

Jedenfalls will man ja so ode rso wieder sauber dastehen, oder?

Reply:
ja - stimmt. Machst du ein Ticket dafür?

Reply:
Neues Ticket erstellt:
DALA 85: Handle inconsistencies between Dataland and EuroDaT

Reply:
ok